app/controllers/partners/dashboards_controller.rb

```diff
@@ -24,6 +24,11 @@ def show
 
       @families = @partner.families
       @children = @partner.children
+
+      @broadcast_announcements = BroadcastAnnouncement.all.select { |announcement|
+        Role.find(announcement.user_id).name == 'org_user' &&
+          (announcement.expiry.nil? || announcement.expiry.strftime('%y %m %d') >= DateTime.now.strftime('%y %m %d'))
```

Comment:
A couple of issues here:

*   Again, it's loading the whole announcement table into memory - the conditions should be part of a `where` clause.
*   This checks that the announcer is an `org_user` but not that the announcer is actually the owning organization.